nf_core/modules/test_yml_builder.py

```diff
@@ -340,7 +340,7 @@ def run_tests_workflow(self, command):
             raise UserWarning(f"Error running test workflow: {e}")
         else:
             log.info("Test workflow finished!")
-            log.debug(rich.markup.escape(nfconfig_raw))
+            log.debug(rich.markup.escape(nfconfig_raw.decode("utf-8")))
```

Comment:
I'd be slightly worried that this could then crash if the returned variable is a string and not bytes (which presumably it has been at some point for this code to get here?)

In MultiQC I have done the following pattern which I think should work for both:

```suggestion
            log.debug(rich.markup.escape(nfconfig_raw.encode("utf-8", "ignore").decode("utf-8")))
```

This first encodes a string to utf-8 but ignores anything it can't encode (like something that is already encoded), _then_ decodes when you're sure it's encoded.

Don't know if there's a nicer way..

Reply:
What do you think about using try/except? 
I also like the suggestion above, but maybe it's faster if we don't encode when it's a sring?

```suggestion
            try:
                        log.debug(rich.markup.escape(nfconfig_raw))
            except TypeError:
                        log.debug(rich.markup.escape(nfconfig_raw.decode("utf-8")))
```

Reply:
Yeah that works too 👍🏻 I doubt there will be a noticeable speed difference, but this is probably clearer / more pythonic.